src/Microsoft.Azure.WebJobs.Host/Triggers/TriggeredFunctionBinding.cs

```diff
@@ -48,16 +48,15 @@ internal class TriggeredFunctionBinding<TTriggerValue> : ITriggeredFunctionBindi
         private async Task<IReadOnlyDictionary<string, IValueProvider>> BindCoreAsync(ValueBindingContext context, object value, IDictionary<string, object> parameters)
         {
             Dictionary<string, IValueProvider> valueProviders = new Dictionary<string, IValueProvider>();
+            ITriggerData triggerData = null;
             IValueProvider triggerProvider;
-            IReadOnlyDictionary<string, object> bindingData;
 
             IValueBinder triggerReturnValueProvider= null;
             try
             {
-                ITriggerData triggerData = await _triggerBinding.BindAsync(value, context);
+                triggerData = await _triggerBinding.BindAsync(value, context);
                 triggerProvider = triggerData.ValueProvider;
-                bindingData = triggerData.BindingData;
-                triggerReturnValueProvider = (triggerData as TriggerData)?.ReturnValueProvider;
```

Comment:
ReturnValueProvider should have been added as a separate interface, not a property on TriggerData, given that the extensibility interfaces aren't based on the TriggerData type